frontend/src/components/features/github/github-repo-selector.tsx

```diff
@@ -16,24 +17,36 @@ export function GitHubRepositorySelector({
 }: GitHubRepositorySelectorProps) {
   const { data: config } = useConfig();
   const [selectedKey, setSelectedKey] = React.useState<string | null>(null);
+  const [searchQuery, setSearchQuery] = React.useState<string>("");
+  const [searchedRepos, setSearchedRepos] = React.useState<GitHubRepository[]>(
+    [],
+  );
+
+  React.useEffect(() => {
+    const searchPublicRepo = async () => {
+      const repos = await searchPublicRepositories(searchQuery);
+      setSearchedRepos(repos);
+    };
+
+    const debounceTimeout = setTimeout(searchPublicRepo, 300);
+    return () => clearTimeout(debounceTimeout);
+  }, [searchQuery]);
```

Comment:
Could outsource logic into a `useDebounce` hook

Reply:
Openhands fix success summary


The feedback has been successfully incorporated. The AI created a new custom hook called `useDebounce` and moved the debouncing logic out of the component into this reusable hook. This is exactly what was requested in the feedback, and it improves code reusability and separation of concerns. The hook was created in a new file at `/workspace/frontend/src/hooks/use-debounce.ts`, making it available for use across the application where needed.

This change follows React best practices by extracting common functionality into a custom hook, making the code more maintainable and reusable. The implementation also maintains all the existing functionality while improving the code structure.